### PR TITLE
Implementing Market and product page

### PR DESCRIPTION
**Pull request by MADahab**
-adding product info that has id name price and preview picture

-Market with name search and category feature

-product page

### CODE REVIEW

File: src/components/market-components/SingleProductCard.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import React, { useState, useEffect } from 'react';
+import { observer } from 'mobx-react-lite';
+import { Link, useNavigate } from 'react-router-dom'; // Import Link for navigation
+
+const SingleProductCard: React.FC<{ product: { id: number, title: string, price: string, category: string, prevPic: string } }> = ({ product }) => {
```
Comment (amrAaboelnaga) [inline src/components/market-components/SingleProductCard.tsx]: put in interface

File: src/components/whiteboard-components/productInfoBox.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>) => {
+        const { name, value } = e.target;
+        whiteBoardStore.setProductInfo({
+            ...whiteBoardStore.productInfo,
+            [name]: value
+        });
+    };
+
+    return (
+        <div style={styles.productInfoInputCont}>
```
Comment (amrAaboelnaga) [inline src/components/whiteboard-components/productInfoBox.tsx:19]: classnames for easier tracking from inspect

File: src/pages/SingleProductPage.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      const productSpec = products.find(product => product.productInfo.id === productId);
+      if (productSpec) {
+        setProductJsonSpec(productSpec.jsonSpecs);
+      } else {
+        console.error(`Product with id ${productId} not found in products array.`);
+      }
+    } else {
+      console.error(`Product with id ${productId} not found in productInfosOnly array.`);
+    }
+  };
```
Comment (amrAaboelnaga) [inline src/pages/SingleProductPage.tsx]: make a handler for this

File: src/pages/SingleProductPage.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+  const useThisTemplate = () => {
+    if (productInfoOnly && productJsonSpec) {
+      whiteBoardStore.setProductInfo(productInfoOnly);
+      whiteBoardStore.setJsonSpecs(productJsonSpec);
+      navigate('/demo');
+    }
+  };
```
Comment (amrAaboelnaga) [inline src/pages/SingleProductPage.tsx]: if manipulating store values put in handlers

File: src/components/market-components/ProductsCont.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
```
Comment (amrAaboelnaga) [inline src/components/market-components/ProductsCont.tsx]: introduce pagination and limit rendering count

File: src/handlers/marketHandlers.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
+import { productInfosOnly } from '../assets/Examples/backEnd';
```
Comment (amrAaboelnaga) [inline src/handlers/marketHandlers.ts]: switch to the agreed format:


import { rootStore } from './path_to_rootStore';

export const useHandler = () => {
    const { whiteBoardStore } = rootStore;
    
    function test1() {
        whiteBoardStore.blabla = blabla;
    }
    
    return { test1 };
};

File: src/stores/marketStore.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
+import { makeAutoObservable, runInAction } from 'mobx';
+import { fetchProductsData } from '../handlers/marketHandlers';
```
Comment (amrAaboelnaga) [inline src/stores/marketStore.ts]: do not put import handler in the store.

File: src/components/market-components/SingleProductCard.tsx (hunk 1 above)
Comment (MADahab) [reply src/components/market-components/SingleProductCard.tsx]: done

File: src/components/whiteboard-components/productInfoBox.tsx (hunk 2 above)
Comment (MADahab) [reply src/components/whiteboard-components/productInfoBox.tsx:19]: waiting to know the approach required.

File: src/pages/SingleProductPage.tsx (hunk 3 above)
Comment (MADahab) [reply src/pages/SingleProductPage.tsx]: done

File: src/pages/SingleProductPage.tsx (hunk 4 above)
Comment (MADahab) [reply src/pages/SingleProductPage.tsx]: done

File: src/components/market-components/ProductsCont.tsx (hunk 5 above)
Comment (MADahab) [reply src/components/market-components/ProductsCont.tsx]: done

File: src/handlers/marketHandlers.ts (hunk 6 above)
Comment (MADahab) [reply src/handlers/marketHandlers.ts]: done, old and new files

File: src/stores/marketStore.ts (hunk 7 above)
Comment (MADahab) [reply src/stores/marketStore.ts]: done

**Pull request merged**